BBB_cci_mpf/test/test-mpf/test_mem_perf/sw/test_mem_perf.h

```diff
@@ -132,20 +132,38 @@ class TEST_MEM_PERF : public CCI_TEST
         {
             vl0_wr_hit_rate = round((1000.0 * stats.write_cache_line_hits) / stats.vl0_wr_lines);
         }
-
-        os << boost::format("%.1f") % ((double(stats.read_lines) * CL(1) / 0x40000000) / stats.run_sec) << " "
-           << boost::format("%.1f") % ((double(stats.write_lines) * CL(1) / 0x40000000) / stats.run_sec) << " "
-           << stats.vl0_rd_lines + stats.vl0_wr_lines << " "
-           << stats.vh0_lines << " "
-           << stats.vh1_lines << " "
-           << vl0_rd_hit_rate << " "
-           << vl0_wr_hit_rate << " "
-           << stats.read_max_inflight_lines << " "
-           << stats.read_average_latency << " "
-           << stats.write_max_inflight_lines << " "
-           << stats.write_average_latency << " "
-           << boost::format("%.4f") % (100.0 * double(stats.read_almost_full_cycles) / double(stats.actual_cycles)) << " "
-           << boost::format("%.4f") % (100.0 * double(stats.write_almost_full_cycles) / double(stats.actual_cycles)) << " ";
+  
+        if (stats.actual_cycles == 0)
+        {
```

Comment:
If actual cycles is 0 then just print a warning message. No need to replicate the whole thing.

Reply:
no need to print other stats? just warning saying actual cycles is zero. I guess other stats might me meaningless but you are printing stats from the cpp code when you see actual cycles is 0, do we need to print stats at all?

Reply:
If actual cycles is 0 then nothing interesting happened. As you might guess from the code in its present form, that never happens.

Reply:
ok , i will just remove print of stats from cpp file instead of changing header file , and print warning in cpp file